File: Problem/By section/Math/Problem 168/problem_168.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    string convertToTitle(int columnNumber) {
        string result  = "";
        
        while(columnNumber) {
            char add_letter = (--columnNumber) % 26 + 'A'; //every time you need decrement by 1, because all encoding is a right shift of [0 - 25]
            columnNumber /= 26;
            result = add_letter + result;
        }
        
        return result;
    }
};

int main(int argc, char** argv) {
    Solution s;
    
    return 0;
}

File: Problem/By section/Math/Problem 168/problem_168.java
class Solution {
    // Time: O(n), Space: O(1)
    public String convertToTitle(int n) {
        if( n == 0) return "";
        // Because it starts from A, which is 1,
        // we have to decrement 1 for each time
        n --;
        return convertToTitle(n / 26) + (char)(n % 26 + 'A');
    }
}

public class problem_168{
    public static void main(String[] args) {
        
    }
}